builtin/providers/aws/provider.go

```diff
@@ -337,6 +348,14 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		Insecure:         d.Get("insecure").(bool),
 	}
 
+	hash, err := hashstructure.Hash(config, nil)
+	if err != nil {
+		return nil, err
+	}
+	if client, ok := cachedClients[hash]; ok {
+		return client, nil
+	}
```

Comment:
@jen20 in discussing this with you in person, you suggested that this mapping was unnecessary, because each distinct AWS config (one per alias) will be in its own process already. Am I remembering that correctly?